kube-core/src/version.rs

```diff
@@ -182,40 +185,40 @@ impl Version {
         }
     }
 
-    /// An [`Ord`] for `Version` that prefers the latest version, even if it is a prerelease
+    /// An [`Ord`] for `Version` that prefers the semantically greatest release (even if it's a pre-release)
```

Comment:
Not sure what K8s API group versions (which only have major versions) have to do with Semver (which is about how to decide when to cut a major/minor/patch release).

That said, I can see the ambiguity problem with `latest` too...

Reply:
Yeah, that's why i went with the word "semantic" to invoke the way semver orders versions (which is the same as here), without it necessarily it being semver.

I do think there's room for improvement in this name though.

Ultimately we need to make it feel like the "lateness of the main line of releases", which is a tricky thing to do briefly.

- we could use synonyms of "progress", like "evolution", or "procession"
- we could maybe use far synonyms of "main" like "lineage" or "origin" (then desc/asc sort are how far / close things are down the main lineage tree)
- synonyms of newness / modernity suffers from same ambiguity as lateness/latest

Reply:
I tried to come up with some synonyms around this and found it hard to wrangle things like `lineage` / `origin` / `generation` and have gone for the perhaps simpler to understand `distance` in that: the greater distance between two versions, the farther they are apart (regardless of when they were released).

it hand-waves a bit how it goes into distance "semantic version distance", but i feel this probably aligns with what people expect: higher initial numbers first, then release specifiers below, then lower initial numbers below even that.

Reply:
Oh, `generation` sounds like the best proposal so far, since that's basically what the major version represents (and within a single major version the ordering should be obvious anyway)!

It's also an absolute measure (as opposed to `distance`), and I even kind of like that it's a bit more obscure than `priority` (which is probably the first choice that we *should* be pushing people towards unless they're specifically after something else).

Reply:
Ok, I that's fair, it's ultimately a single number. `generation` arguably hand-waves "what specifiers mean" here, but `distance` ultimately ends up hand-waving "what distance means" so this is probably better. happy to use `generation` :+1: